addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/ExternalRedirectScanRule.java

```diff
@@ -476,11 +482,157 @@ private static RedirectType isRedirected(String payload, HttpMessage msg) {
     }
 
     private static boolean isRedirectPresent(Pattern pattern, String value) {
-        Matcher matcher = pattern.matcher(value);
+        // We know the payload is present.
+        // Ensure the value has something we're likely interested in before dealing with comments
+        if (JS_PRE_CHECKS.stream().noneMatch(chk -> StringUtils.containsIgnoreCase(value, chk))) {
```

Comment:
I'd suggest keeping the previous check (even if that means matching twice), instead of adding additional checks which must be kept in sync with the patterns.

Reply:
The `pattern` check. Yes, the `JS_PRE_CHECKS`.

Reply:
So do `matcher.find` both before and after?

Reply:
Right.

Reply:
Done & done

Reply:
Done